src/reactA11yMediaCaptionsRule.ts

```diff
@@ -0,0 +1,128 @@
+import * as ts from 'typescript';
+import * as Lint from 'tslint';
+import * as tsutils from 'tsutils';
+
+import { ExtendedMetadata } from './utils/ExtendedMetadata';
+import { getAllAttributesFromJsxElement } from './utils/JsxAttribute';
+
+const VIDEO_ELEMENT_NAME: string = 'video';
+const AUDIO_ELEMENT_NAME: string = 'audio';
+const TRACK_ELEMENT_NAME: string = 'track';
+const KIND_ATTRIBUTE_NAME: string = 'kind';
+const CAPTIONS_KIND_NAME: string = 'captions';
+const DESCRIPTION_KIND_NAME: string = 'description';
+const NO_CAPTIONS_ERROR_MESSAGE: string = 'Video and audio elements must have a captions track';
+const NO_AUDIO_DESCRIPTION_MESSAGE: string = 'Video elements must have an audio description track';
+
+export class Rule extends Lint.Rules.AbstractRule {
+    public static metadata: ExtendedMetadata = {
+        ruleName: 'react-a11y-media-captions',
+        type: 'functionality',
+        description: 'Enforce that video and audio elements have captions and descriptions.',
+        options: null, // tslint:disable-line:no-null-keyword
+        optionsDescription: '',
+        typescriptOnly: false,
+        issueClass: 'Non-SDL',
+        issueType: 'Error',
+        severity: 'Important',
+        level: 'Opportunity for Excellence',
+        group: 'Accessibility'
+    };
+
+    public apply(sourceFile: ts.SourceFile): Lint.RuleFailure[] {
+        return sourceFile.languageVariant === ts.LanguageVariant.JSX ? this.applyWithFunction(sourceFile, walk) : [];
+    }
+}
+
+function walk(ctx: Lint.WalkContext<void>) {
+    function cb(node: ts.Node): void {
+        if (tsutils.isJsxSelfClosingElement(node) && (isVideoElement(node.tagName) || isAudioElement(node.tagName))) {
+            ctx.addFailureAtNode(node, NO_CAPTIONS_ERROR_MESSAGE);
+        }
+        if (tsutils.isJsxElement(node) && (isVideoElement(node.openingElement.tagName) || isAudioElement(node.openingElement.tagName))) {
+            if (!containsTrackElementWithSpreadAttributes(node.openingElement)) {
+                validateMediaType(node.openingElement, ctx);
+                return;
+            }
+        }
+        return ts.forEachChild(node, cb);
+    }
+    return ts.forEachChild(ctx.sourceFile, cb);
+}
+
+function isVideoElement(tagName: ts.JsxTagNameExpression): boolean {
+    const elementName = tagName.getText();
+    return elementName === VIDEO_ELEMENT_NAME;
+}
+
+function isAudioElement(tagName: ts.JsxTagNameExpression): boolean {
+    const elementName = tagName.getText();
+    return elementName === AUDIO_ELEMENT_NAME;
+}
+
+function getAttributeText(attribute: ts.JsxAttribute): string | undefined {
```

Comment:
Looks like this function can be replaced with existing utility `getStringLiteral` from [`src/utils/JsxAttribute.ts`](https://github.com/microsoft/tslint-microsoft-contrib/blob/34f334767984637a70fe958f176d2ffc8b92eb58/src/utils/JsxAttribute.ts#L33)

Which has similar signature and results I think